Lib/test/test_socketserver.py

```diff
@@ -492,6 +492,37 @@ def shutdown_request(self, request):
         self.assertEqual(server.shutdown_called, 1)
         server.server_close()
 
+    def test_no_accept_after_shutdown(self):
+        rejected = False
+        s = socketserver.TCPServer((HOST, 0), socketserver.StreamRequestHandler)
+
+        thread_serve = threading.Thread(
+            name='MyServer serving',
+            target=s.serve_forever)
+        thread_serve.daemon = True  # In case this function raises.
+        thread_serve.start()
+
+        thread_shutdown = threading.Thread(
+            name='MyServer shutdown',
+            target=s.shutdown)
+        thread_shutdown.daemon = True  # In case this function raises.
+        thread_shutdown.start()
+
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(0.1)
```

Comment:
It seems like your test rely on time for synchronization. I feel that it will fail if the sytem is busy or the threads are ordered differently depending on the OS :-(

I would either prefer to remove the test or write a test with better synchronization between the 3 threads.

cc @pablogsal

Reply:
The unit test was mainly to demonstrate there was an issue and how it could be triggered.

I hardly see how I could make the unit test not time based.

For the part
```python
sock.settimeout(0.1)
...
try:
    sock.recv(1024)
except socket.timeout:
    rejected = True
```
I could indeed maybe find another heuristic to determine the request was not accepted, but then remains the ```poll_interval``` timeout of the method
```python
def serve_forever(self, poll_interval=0.5):
```

If I increase this, this will take longer for the server to shutdown, and therefore the test will take time for an unnecessary reason.

I have no problem the unit test not to be merged. As explained, this was mainly to demonstrate the issue. Maybe it can be used as a reference in this pull request in case someone would like to know why this two lines patch was added.

I am opened to your suggestions if you have an idea to make this not time-based. Otherwise let me know and I will add a commit to remove the unit test.

Reply:
Indeed, I was able to make this test fail several times in our most slow buildbots :(

Also, shouldn't this be decorated with `@reap_threads`?

Reply:
Please just remove the test. The fix is trivial, but I don't want to have to debug race condition in this new functional test.

Reply:
I fully agree. I just removed the unit test case.